pkg/objects/swift.go

```diff
@@ -253,32 +253,42 @@ func (s *SwiftLocation) ObjectAtPath(path string) *schwift.Object {
 }
 
 //ListAllFiles implements the Source interface.
-func (s *SwiftLocation) ListAllFiles() ([]FileSpec, *ListEntriesError) {
-	return s.listFiles("", true)
+func (s *SwiftLocation) ListAllFiles(out chan<- FileSpec) *ListEntriesError {
+	objectPath := s.ObjectNamePrefix
+	if objectPath != "" && !strings.HasSuffix(objectPath, "/") {
+		objectPath += "/"
+	}
+	logg.Debug("listing objects at %s/%s recursively", s.ContainerName, objectPath)
+
+	iter := s.Container.Objects()
+	iter.Prefix = objectPath
+	iter.Delimiter = "/"
```

Comment:
When listing recursively, `iter.Delimiter` must be unset. The original code has this line within `if !recursively { ... }`.